### ISSUE TITLE
#notanissuebutarequest #energymeter

### ISSUE THREAD

**Issue by Tokn59** [DanielOgorchock/ST_Anything#163]
Hi, any chance the following child devices can be added : 

- current (in Amps) to be used in energy monitors
- energy (in Kwh) to be used in energy monitors

Would be great. Sorry to post as issue but did not know where else to "request"

Kind regards,

Tony

**Comment by ogiewon** [comment]
I can add an Energy Meter child device, but I don’t have anything yet on the Arduino side to measure Energy.  Any ideas?  Or are you rolling you’re own sketch and just using my Device Handlers?

As for Current, I don’t see a corresponding SmartThings Capability for the measurement of Current. I typically don’t create custom device types as I believe in trying to stay aligned with official device Capabilities. 

https://docs.smartthings.com/en/latest/capabilities-reference.html#introduction

**Comment by Tokn59** [comment]
I use hte PZEM-OOU (cheap) energy sensor. It communicates via RS232. But in my sketch I only use the Smartthings Library to send data to ST manually. 

For current I could maybe use the "generic" child device ?

Thanks !!!!

**Comment by Tokn59** [comment]
Hello, any nws on this please ?

Kind Regards,

Tony

**Comment by ogiewon** [comment]
Tony,

I have added the Child Energy Meter groovy Device Handler for you.  I have also made the necessary changes in the Parent Device Handler to support it.

In order to use it, your sketch will need to send data in the form of "energy1 12.34".  Since ST_Anything does not have support for the device you're using, I am going to assume that you'll be writing your own sketch and using my "SmartThings..." communications libraries to send the data after reading it periodically (i.e. not every time through the loop() routine).  For power, make sure to send it as "power1 12.34".  The name and case is critical for the parent groovy code to parse it correctly.  I hope this gets you started.
